src/test/java/uk/gov/companieshouse/api/accounts/interceptor/TransactionInterceptorTest.java

```diff
@@ -0,0 +1,96 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionStatus;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+public class TransactionInterceptorTest {
+
+    @InjectMocks
+    private TransactionInterceptor transactionInterceptor;
+
+    @Mock
+    private TransactionManager transactionManagerMock;
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private HttpServletResponse httpServletResponse;
+
+    @BeforeEach
+    public void setUp() {
+        Map<String, String> pathVariables = ImmutableMap.of("transactionId", "5555");
+
+        when(httpServletRequest.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE))
+                .thenReturn(pathVariables);
+        when(httpServletRequest.getHeader("X-Request-Id")).thenReturn("1111");
+
+        httpServletResponse.setContentType("text/html");
+    }
+
+    @Test
+    @DisplayName("Tests the interceptor with an existing transaction that is open")
+    public void testPreHandleWithOpenTransaction() {
+        when(transactionManagerMock.getTransaction(anyString(), anyString()))
+                .thenReturn(createOpenDummyTransaction(true));
+
+        assertTrue(transactionInterceptor.preHandle(httpServletRequest, httpServletResponse, new Object()));
+    }
+
+    @Test
+    @DisplayName("Tests the interceptor with an existing transaction that is closed")
+    public void testPreHandleWithClosedTransaction() {
+        when(transactionManagerMock.getTransaction(anyString(), anyString()))
+                .thenReturn(createOpenDummyTransaction(false));
+
+        assertFalse(transactionInterceptor.preHandle(httpServletRequest, httpServletResponse, new Object()));
+    }
+
+    @Test
+    @DisplayName("Tests the interceptor with a non-existing transaction")
+    public void testPreHandleWithNonExistingTransaction() {
+        when(transactionManagerMock.getTransaction(anyString(), anyString()))
+                .thenThrow(new HttpClientErrorException(HttpStatus.NOT_FOUND));
+
+        assertFalse(transactionInterceptor.preHandle(httpServletRequest, httpServletResponse, new Object()));
```

Comment:
I did do that bit, but I had great trouble verifying the response codes. We cannot set properties on the mock response object, but also cannot pass through a response object because it is an interface. Am open to suggestions.

Reply:
Import:
```
import static org.mockito.Mockito.verify;
```

Verify with:
```
verify(httpServletResponse).setStatus(HttpStatus.NOT_FOUND.value());
```

Reply:
done